Initial commit (#5060)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/terraform-configure-device-signin-standards/main/index.md

The following code shows a typical iOS device assurance policy:

```hcl
resource okta_policy_device_assurance_ios ios_example {
  name = "iOS example"
  os_version = "12.4.5"
  jailbreak = false
  screenlock_type = toset(["BIOMETRIC"])
}
```

### macOS

Create a macOS device assurance policy:

1. Add the `okta_policy_device_assurance_macos` resource to your configuration.

1. Set the `name` argument to a user-visible name for this policy.

1. Set arguments for different signals. The following are the most commonly checked signals. For the complete list of signals, see the [Terraform provider documentation for this resource](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/policy_device_assurance_macos).

    * `os_version`: The minimum operating system version for a device.

    * `disk_encryption_type`: The set of required encryption settings for the device. The only option is `ALL_INTERNAL_VOLUMES` ( all internal volumes are encrypted).

    * `screenlock_type`: A set of required screen lock actions and behaviors.  The supported options are `BIOMETRIC` (biometric locking) and `PASSCODE` (passcode locking).

> Your policy can also check signals on devices from a supported third-party signal provider, such as the Google Chrome Device Trust Connector in the Chrome browser. Specify its device signals in your Terraform code with the fields with the prefix `tpsp_`. For example, `tpsp_os_version` instead of `os_version`.  Okta for device assurance for ChromeOS requires using [Workforce Identity Adaptive Multi-factor Authentication (AMFA)](https://www.okta.com/learn/adaptive-mfa/). Developer Edition Okta orgs don't support the Google Chrome Device Trust Connector by default. If you're a paying customer, contact Okta developer support to enable third-party signal support on a Developer Edition Okta org.

Reviewer: ```suggestion
Your policy can also check signals on devices from a supported third-party signal provider, such as the Google Chrome Device Trust Connector in the Chrome browser. Specify its device signals in your Terraform code with the fields with the prefix `tpsp_`. For example, `tpsp_os_version` instead of `os_version`.  Okta for device assurance for ChromeOS requires using [Workforce Identity Adaptive Multi-factor Authentication (AMFA)](https://www.okta.com/learn/adaptive-mfa/). Developer Edition orgs don't support the Google Chrome Device Trust Connector by default. If you're a paying customer, contact Okta developer support to enable third-party signal support on a Developer Edition org.
```

The following code shows a typical macOS device assurance policy:

```hcl
resource okta_policy_device_assurance_macos macos_example {
  name = "macOS example"
  os_version = "12.4.6"
  disk_encryption_type = toset(["ALL_INTERNAL_VOLUMES"])
  secure_hardware_present = true
  screenlock_type = toset(["BIOMETRIC", "PASSCODE"])
}
```

### Windows

Create a Windows device assurance policy:

1. Add the `okta_policy_device_assurance_windows` resource to your configuration.

1. Set its `name` argument to a user-visible name for this policy.

1. Set arguments for different signals. The following are the most commonly checked signals. For the complete signal list, see the [provider documentation for this resource](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/policy_device_assurance_windows).

    * `os_version`: The minimum operating system for a device.

>     * `disk_encryption_type`: The set of required encryption settings for the device. The only option is `ALL_INTERNAL_VOLUMES` ( all internal volumes are encrypted).

Reviewer: ```suggestion
    * `disk_encryption_type`: The set of required encryption settings for the device. The only option is `ALL_INTERNAL_VOLUMES` (all internal volumes are encrypted).
```

    * `screenlock_type`: A set of required screen lock actions and behaviors. The supported options are `BIOMETRIC` (biometric locking) and `PASSCODE` (passcode locking).

> Your policy can also check signals on devices from a supported third-party signal provider, such as the Google Chrome Device Trust Connector in the Chrome browser. Specify its device signals in your Terraform code with the fields with the prefix `tpsp_`. For example, `tpsp_os_version` instead of `os_version`.  Okta for device assurance for ChromeOS requires using [Workforce Identity Adaptive Multi-factor Authentication (AMFA)](https://www.okta.com/learn/adaptive-mfa/). Developer Edition Okta orgs don't support the Google Chrome Device Trust Connector by default. If you're a paying customer, contact Okta developer support to enable third-party signal on a Developer Edition Okta org.

Reviewer: ```suggestion
Your policy can also check signals on devices from a supported third-party signal provider, such as the Google Chrome Device Trust Connector in the Chrome browser. Specify its device signals in your Terraform code with the fields with the prefix `tpsp_`. For example, `tpsp_os_version` instead of `os_version`.  Okta for device assurance for ChromeOS requires using [Workforce Identity Adaptive Multi-factor Authentication (AMFA)](https://www.okta.com/learn/adaptive-mfa/). Developer Edition orgs don't support the Google Chrome Device Trust Connector by default. If you're a paying customer, contact Okta developer support to enable third-party signal on a Developer Edition org.
```

The following code shows a typical Windows device assurance policy:

```hcl
resource okta_policy_device_assurance_windows windows_example {
  name = "Windows example"
  os_version = "12.4.6"
  disk_encryption_type = toset(["ALL_INTERNAL_VOLUMES"])
  secure_hardware_present = true
  screenlock_type = toset(["BIOMETRIC", "PASSCODE"])
}
```

### ChromeOS

> Some Okta device assurance policies support ChromeOS configurations. Okta for device assurance for ChromeOS requires using [Workforce Identity Adaptive Multi-factor Authentication (AMFA)](https://www.okta.com/learn/adaptive-mfa/). Developer Edition Okta orgs don't support the Google Chrome Device Trust Connector by default. If you're a paying customer, contact Okta developer support to enable ChromeOS support on a Developer Edition Okta org.

Reviewer: ```suggestion
Some Okta device assurance policies support ChromeOS configurations. Okta for device assurance for ChromeOS requires using [Workforce Identity Adaptive Multi-factor Authentication (AMFA)](https://www.okta.com/learn/adaptive-mfa/). Developer Edition orgs don't support the Google Chrome Device Trust Connector by default. If you're a paying customer, contact Okta developer support to enable ChromeOS support on a Developer Edition org.
```

    * `tpsp_os_version`: The minimum operating system for a device.

    * `tpsp_disk_encrypted`: Checks if the disk is encrypted.

    * `tpsp_screen_lock_secured`: Specifies if the screen lock is secured.

The following code shows a typical ChromOS device assurance policy:

```hcl
resource okta_policy_device_assurance_chromeos example{
  name = "example chromeos"
  tpsp_allow_screen_lock = true
  tpsp_browser_version = "15393.27.0"
  tpsp_builtin_dns_client_enabled = true
  tpsp_chrome_remote_desktop_app_blocked = true
  tpsp_disk_encrypted = true
  tpsp_os_firewall = true
  tpsp_os_version = "10.0.19041.1110"
  tpsp_realtime_url_check_mode = true
  tpsp_screen_lock_secured = true
}
```

## Use device assurance for authentication

To use device assurance, add one or more device assurance policies to an authentication policy.

There are two approaches to defining an authentication policy:

> * Use the built-in default authentication policy representing your Okta app, such as an OAuth web app. In this case, use a Terraform data source to find the ID for the default authentication policy and then add a rule referencing that policy. For an example, see this [policy rule example in the Terraform git repository](https://github.com/okta/terraform-provider-okta/blob/master/examples/resources/okta_app_signon_policy_rule/basic.tf).

Reviewer: ```suggestion
* Use the default authentication policy representing your Okta app, such as an OAuth web app. In this case, use a Terraform data source to find the ID for the default authentication policy and then add a rule referencing that policy. For an example, see this [policy rule example in the Terraform git repository](https://github.com/okta/terraform-provider-okta/blob/master/examples/resources/okta_app_signon_policy_rule/basic.tf).
```

> * Create an authentication policy for your Okta app and set the policy ID in a field in your app resource in Terraform.

Reviewer: ```suggestion
* Create an authentication policy for your app and set the policy ID in a field in your app resource in Terraform.
```

The high-level steps to enforce device minimum standards:

1. Create device assurance policies for your supported platforms.

> 1. If you don't already have one, create an Okta application for which you want access control, such as an OAuth web app. See the provider for resources with the name `okta_app_<type_name>`.

Reviewer: ```suggestion
1. If you don't already have one, create an app for which you want access control, such as an OAuth web app. See the provider for resources with the name `okta_app_<type_name>`.
```

1. Create an authentication policy resource of type [okta_app_signon_policy](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_signon_policy).

1. Create an authentication policy rule of type [okta_app_signon_policy_rule](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_signon_policy_rule) that specifies your device assurance policies. Each rule references its associated policy in its argument `policy_id`. One rule can reference multiple device assurance policies using a list of their IDs.

> 1. Attach the authentication policy to your Okta application by setting the app's policy ID in the `authentication_policy` field.

Reviewer: ```suggestion
1. Attach the authentication policy to your app by setting the policy ID in the `authentication_policy` field.
```

> The following example creates an OAuth application using an `okta_app_oauth` resource and then adds an authentication policy (a sign-on policy) that contains two policy rules:

Reviewer: ```suggestion
The following example creates an OAuth app using an `okta_app_oauth` resource and then adds an authentication policy (a sign-on policy) that contains two policy rules:
```
